compiler/luci/pass/src/RemoveNoEffectSlicePass.cpp

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/RemoveNoEffectSlicePass.h"
+
+#include <luci/IR/CircleNodes.h>
+
+namespace
+{
+
+/// @brief Return value in CircleConst.
+/// @details Return -2 if CircleConst is nullptr or not valid shape, otherwise return value in
+///          position on CircleConst with int64 format.
+///          Begin & size must be larger than or equal to -1, so -2 is invalid value.
+int64_t value_from_circle_const(const luci::CircleConst *node, uint32_t idx)
+{
+  if (node == nullptr || node->rank() != 1 || node->dim(0).value() <= idx)
```

Comment:
`node == nullptr` will never happen because `size_const` and `begin_const` is already checked by line 42 and line 46.

If some invalid situation is occurred, throwing runtime error would be better because it should not be happened.

Reply:
And one more thing, returning -2 seems weird because `value_from_circle_const` implies **the value in the const node**. However -2 is not in the node.

Reply:
Okay I will change to throw error on this case. and also invalid Data type case.